File: core/src/main/java/org/nedezkiiyasen/uha/core/model/Event.java
package org.nedezkiiyasen.uha.core.model;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.sql.Date;
import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "events")
@Getter
@Setter
public class Event extends RepositoryItem {
    @ManyToMany(fetch = FetchType.EAGER)
    @JoinTable(
            name = "eptituders_events",
            joinColumns = { @JoinColumn(name = "event_id") },
            inverseJoinColumns = { @JoinColumn(name = "eptituder_id") }
    )
    private Set<Eptituder> eptituders = new HashSet<>();
    private String name;
    private String ico;
    private String description;
    private Date date;
    private Integer points;
    private String status;
    private String rank;
}

File: spa/src/main/ng/src/app/event.ts
import {Eptituder} from "./eptituder";

export class Event {
  id: number;
  name: string;
  ico: string;
  description: string;
  date: Date;
  points: number;
  status: string;
  rank: string;
  eptituders: Eptituder[];
}
